src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -56,6 +70,28 @@ public static IMvcCoreBuilder AddRazorViewEngine(this IMvcCoreBuilder builder)
         }
 
         /// <summary>
+        /// Registers discovered tag helpers as services and changes the existing <see cref="ITagHelperActivator"/>
+        /// for an <see cref="ServiceBasedTagHelperActivator"/>.
+        /// </summary>
+        /// <param name="builder">The <see cref="IMvcCoreBuilder"/> instance this method extends.</param>
+        /// <returns>The <see cref="IMvcCoreBuilder"/> instance this method extends.</returns>
+        public static IMvcCoreBuilder AddTagHelpersAsServices(this IMvcCoreBuilder builder)
```

Comment:
I don't really have a strong feeling either way. It doesn't seem harmful to call `AddRazorViewEngine()` from here, they are all idempotent